Mathlib/Analysis/Normed/Field/Ultra.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Analysis.Normed.Group.Ultra
+
+/-!
+## Sufficient contidition to have an ultrametric norm on a field
+
+This file provides ways of constructing an instance of `IsUltrametricDist` based on
+facts about the existing norm.
+
+## Main results
+
+* `isUltrametricDist_of_forall_norm_natCast_le_one`: a norm in a field is ultrametric if the norm
+  of the image of a natural is less than or equal to one
+
+## Implementation details
+
+The proof relies on a bounded-from-above argument, so the imports need to include `rpow`
+
+## Tags
+
+ultrametric, nonarchimedean
+-/
+open Metric IsUltrametricDist NNReal
+
+namespace IsUltrametricDist
+
+section sufficient
+
+variable {R F : Type*} [NormedDivisionRing R] [NormedField F]
+
+lemma isUltrametricDist_of_forall_norm_add_one_le_max_norm_one
+    (h : ∀ x : R, ‖x + 1‖ ≤ max ‖x‖ 1) : IsUltrametricDist R := by
+  apply isUltrametricDist_of_forall_norm_add_le_max_norm
+  intro x y
+  rcases eq_or_ne y 0 with rfl|hy
+  · simp
+  rw [← div_le_div_right (c := ‖y‖) (by simpa using hy), ← norm_div, add_div, div_self hy,
+      ← max_div_div_right (norm_nonneg _), div_self (by simp [hy]), ← norm_div]
+  exact h _
```

Comment:
```suggestion
    (h : ∀ x : R, ‖x + 1‖ ≤ max ‖x‖ 1) : IsUltrametricDist R := by
  refine isUltrametricDist_of_forall_norm_add_le_max_norm (fun x y ↦ ?_)
  rcases eq_or_ne y 0 with rfl | hy
  · simpa only [add_zero] using le_max_left _ _
  · have p : 0 < ‖y‖ := norm_pos_iff.mpr hy
    simpa only [div_add_one hy, norm_div, div_le_iff₀ p, max_mul_of_nonneg _ _ p.le, one_mul,
      div_mul_cancel₀ _ p.ne'] using h (x / y)
```
(I rejigged the proof to have fewer backwards rewrites, it's completely cosmetic)